docker/Dockerfile

```diff
@@ -1,51 +0,0 @@
-# To build use:
```

Comment:
Pretty much agree with removing the other Dockerfiles.

But this one still has a link in docker hub: https://hub.docker.com/r/postgrest/postgrest.

Maybe we should keep it until we make a new release with the new docker image.

Reply:
Ah, yes - we should definitely keep it then! Does the docker-compose file still have a function?

Reply:
Once we make a release with the new image we'll drop this Dockerfile.

Also, we'll probably need some wording on the new image. Maybe a README inside `./nix/docker`? This could then be copy pasted to the docker hub overview.

I'm reading https://nixos.org/nixpkgs/manual/#sec-pkgs-dockerTools. And I think we could put something there about why the Nix approach allows us to get better reproducibility and a leaner image.

A change like this would also merit an addition in the CHANGELOG.

What do you think?

Reply:
> Ah, yes - we should definitely keep it then! Does the docker-compose file still have a function?

On second thought, I think I can just edit the docker hub overview and point it to a previous commit. No need for restoring the files.

But we would need to have the above README and entry in the CHANGELOG. We have a lot of dockers users, they'll wonder about the change.

Reply:
@steve-chavez Sounds good!

I've added:
* a [README](https://github.com/monacoremo/postgrest/tree/monacoremo-release/nix/docker) for the docker build
* a first draft for the [description on Docker Hub](https://github.com/monacoremo/postgrest/blob/monacoremo-release/nix/release/docker-hub-full-description.md) (tested [here](https://hub.docker.com/repository/docker/monacoremo/postgrest))
* a script in `nix/release/default.nix` that pushes the description to Docker Hub, `postgrest-release-dockerhubdescription`
* and the CHANGELOG entry

Reply:
@monacoremo That's some advanced Docker stuff :).

TIL thanks to your README!. The explanation is perfect and the docker hub description release is extra nice.